### PR TITLE
Fix right sidebar state logic 

### PR DESCRIPTION
**Pull request by alexandrudanpop**
Fixes OPS-1320

## Additional context
- When we add or replace a trigger or a step, the step settings are always shown
- Undo / Redo closes step settings

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)
Deployed to UX
https://www.loom.com/share/2d1ea79dc9ff4904965f3a31c701bcfb

### CODE REVIEW

File: packages/react-ui/src/app/features/builder/flow-version-undo-redo/hooks/flow-version-undo-redo.ts [hunk 1]
```diff
@@ -113,6 +120,8 @@ const useFlowVersionUndoRedo = (): FlowVersionUndoRedo => {
   }, [debouncedProcessQueue]);
 
   const enqueueAction = (action: MoveActionType) => {
+    exitStepSettings();
```
Comment (alexandrudanpop) [inline packages/react-ui/src/app/features/builder/flow-version-undo-redo/hooks/flow-version-undo-redo.ts:123]: When we do undo / redo operations we close the step settings.

**Review by rSnapkoOpenOps — APPROVED** [review]
lgtm
